dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo

```diff
@@ -0,0 +1,105 @@
+within Dynawo.Electrical.Controls.Voltage.StaticVarCompensator;
+
+/*
+* Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model ModeHandling "Static Var Compensator mode calculation and blocking"
+  import Modelica;
+  import Dynawo.Electrical.Controls.Voltage.StaticVarCompensator.Parameters;
+
+  extends Parameters.Params_ModeHandling;
+
+  Modelica.Blocks.Interfaces.RealInput URef(start = URef0) "Voltage reference for the regulation in kV" annotation(
+    Placement(visible = true, transformation(origin = {-120, 60}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -32}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput UPu "Voltage at the static var compensator terminal in p.u (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, 20}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+
+  Modelica.Blocks.Interfaces.RealOutput URefPu(start = URef0 / UNom) "Voltage reference for the regulation in p.u (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanOutput blocked "Wheter the static var compensator is blocked due to very low voltages" annotation(
+    Placement(visible = true, transformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  ModeConnector mode(value(start = Mode0)) "Current mode of the static var compensator" annotation(
+    Placement(visible = true, transformation(origin = {100, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {110, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  ModeConnector mode_auto(value(start = Mode0)) "Mode of the static var compensator when in automatic configuration";
+  ModeConnector mode_manual(value(start = Mode0)) "Mode of the static var compensator when in manual configuration";
+
+  Modelica.Blocks.Interfaces.IntegerInput setMode "Mode selected when in manual configuration" annotation(
```

Comment:
Add manual somewhere in the variable name for clarity purpose

Reply:
fixed, replaced by setModeManual